mbtiles/scripts/cli.py

```diff
@@ -118,99 +134,119 @@ def mbtiles(ctx, files, output, force_overwrite, title, description,
+                ("type", layer_type))
             cur.execute(
-                "INSERT INTO tiles "
-                "(zoom_level, tile_column, tile_row, tile_data) "
-                "VALUES (?, ?, ?, ?);",
-                (tile.z, tile.x, tiley, buffer(contents)))
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("version", "1.1"))
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("description", description))
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("format", img_ext))
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("bounds", "%f,%f,%f,%f" % (west, south, east, north)))
 
             conn.commit()
 
+            # Constrain longitude and latitude bounds.
+            EPS = 1.0e-10
+            west = max(-180 + EPS, west)
+            south = max(-85.051129, south)
+            east = min(180 - EPS, east)
+            north = min(85.051129, north)
+
+            with WarpedVRT(
+                    src, dst_crs='EPSG:3857', src_nodata=src_nodata,
+                    dst_nodata=dst_nodata, resampling=Resampling.bilinear,
+                    num_threads=num_workers) as vrt, warnings.catch_warnings():
+
+                warnings.simplefilter("ignore", category=UserWarning)
+
+                for tile in mercantile.tiles(
+                        west, south, east, north,
+                        list(range(minzoom, maxzoom + 1))):
+
+                    window = vrt.window(*mercantile.xy_bounds(*tile))
+                    bands = vrt.read(
+                        window=window, out_shape=(3, 256, 256),
+                        boundless=True)
+
+                    with MemoryFile() as memfile:
+                        # Silence the dubious 'No such file or directory.'
+                        # errors coming from the VSIL handler.
+                        gdal_log = logging.getLogger('rasterio._gdal')
+                        gdal_log_level = gdal_log.level
+                        gdal_log.setLevel(logging.CRITICAL)
+                        with memfile.open(**base_kwds) as tmp:
+                            tmp.write(bands)
+                        gdal_log.setLevel(gdal_log_level)
+                    data = memfile.read()
+
+                    # Workaround for https://bugs.python.org/issue23349.
+                    if sys.version_info[0] == 2 and sys.version_info[2] < 10:
+                        data[:] = data[-1:] + data[:-1]
+
+                    # MBTiles has a different origin than Mercantile/tilebelt.
+                    tiley = int(math.pow(2, tile.z)) - tile.y - 1
+
+                    # Optional image dump.
+                    if image_dump:
+                        img_name = '%d-%d-%d.%s' % (
+                            tile.x, tiley, tile.z, img_ext)
```

Comment:
Since the slippy name format is z/x/y, it seems reasonable to use that standard here as well. Also perhaps a more useful filename for sorting the images in this output directory.